pytest_mozwebqa/cloud/saucelabs.py

```diff
@@ -7,124 +7,138 @@
-                    "queryString":"%%3Fstart%%3D%%24%%7Bstart%%7D"},\
-                "controls":{\
-                    "mute":false,\
-                    "volume":false,\
-                    "backgroundColor":"rgba(0,0,0,0.7)"}},\
-            "playerId":"player%(session)s",\
-            "playlist":[{\
-                "url":"https://assets.saucelabs.com/jobs/%(session)s/video.flv",\
-                "provider":"streamer",\
-                "autoPlay":false,\
-                "autoBuffering":true}]}' % {'session': session}
-
-        return html.div(html.object(
-            html.param(value='true', name='allowfullscreen'),
-            html.param(value='always', name='allowscriptaccess'),
-            html.param(value='high', name='quality'),
-            html.param(value='#000000', name='bgcolor'),
-            html.param(
-                value=flash_vars.replace(' ', ''),
-                name='flashvars'),
-            width='100%',
-            height='100%',
-            type='application/x-shockwave-flash',
-            data='https://cdn1.saucelabs.com/sauce_skin_deprecated/lib/flowplayer/flowplayer-3.2.17.swf',
-            name='player_api',
-            id='player_api'),
-            id='player%s' % session,
-            style='border:1px solid #e6e6e6; float:right; height:240px;'
-                  'margin-left:5px; overflow:hidden; width:320px')
+name = 'Sauce Labs'
+
+
+def _get_config(option):
+    config = ConfigParser()
+    config.read('setup.cfg')
+    section = 'saucelabs'
+    if config.has_section(section) and config.has_option(section, option):
+        return config.get(section, option)
+
+
+def _credentials():
+    username = os.getenv('SAUCELABS_USERNAME', _get_config('username'))
+    api_key = os.getenv('SAUCELABS_API_KEY', _get_config('api-key'))
+    if username is None:
+        raise pytest.UsageError('Sauce Labs username must be set')
+    if api_key is None:
+        raise pytest.UsageError('Sauce Labs API key must be set')
+    return username, api_key
+
+
+def _tags():
+    tags = _get_config('tags')
+    if tags is not None:
+        return tags.split(',')
+    else:
+        return []
+
+
+def _split_class_and_test_names(nodeid):
+    names = nodeid.split("::")
+    names[0] = names[0].replace("/", '.')
+    names = [x.replace(".py", "") for x in names if x != "()"]
+    classnames = names[:-1]
+    classname = ".".join(classnames)
+    name = names[-1]
+    return (classname, name)
+
+
+def start_driver(item, options, capabilities):
+    from _pytest.mark import MarkInfo
+    tags = _tags() + [m for m in item.keywords.keys() if isinstance(
+        item.keywords[m], MarkInfo)]
+    try:
+        privacy = item.keywords['privacy'].args[0]
+    except (IndexError, KeyError):
+        # privacy mark is not present or has no value
+        privacy = _get_config('privacy')
+
+    if options.browser_name is None:
```

Comment:
I won't add any more comments about `is None` / `is not None`, but assume I did. ;)